src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```
@@ -0,0 +1,319 @@
+   * prevention API.
+   *
+   * @param record a <code>StructuredRecord</code> being passed from the previous stage.
+   * @param emitter a <code>MultiOutputEmitter</code> to emit sensitive or non-sensitive data on different ports.
+   */
+  @Override
+  public void transform(StructuredRecord record, MultiOutputEmitter<StructuredRecord> emitter) throws Exception {
+    String recordString = null;
+    ContentItem contentItem = null;
+
+    if (!config.isEntireRecord()) {
+      recordString = record.get(config.getFieldName()).toString();
+    } else {
+      recordString = StructuredRecordStringConverter.toDelimitedString(record, ",");
+    }
+    try {
+
+      // depending on input schema field object
+      contentItem = ContentItem.newBuilder().setValue(recordString).build();
+
+      String templateName = String.format("projects/%s/inspectTemplates/%s", config.getProject(), config.templateId);
+
+      InspectContentRequest request =
+        InspectContentRequest.newBuilder()
+                             .setParent(ProjectName.of(config.getProject()).toString())
+                             .setInspectTemplateName(templateName)
+                             .setItem(contentItem)
+                             .build();
+
+      metrics.count("dlp.requests.count", 1);
+      InspectContentResponse response = client.inspectContent(request);
+      InspectResult result = response.getResult();
+
+      metrics.count("dlp.requests.success", 1);
+      if (result.getFindingsList().size() > 0) {
+        emitter.emit(SENSITIVE_PORT, record);
+        return;
+      }
+
+      emitter.emit(NON_SENSITIVE_PORT, record);
+
+    } catch (Exception e) {
+      metrics.count("dlp.requests.fail", 1);
+      if (e instanceof ResourceExhaustedException) {
+        LOG.error(
+          "Failed due to DLP rate limit, please request more quota from DLP: https://cloud.google"
+            + ".com/dlp/limits#increases");
+        throw e;
```

Lets not log and throw. we can just throw the exception. Or create another exception with the message you want to add and then throw it

---

done